### PR TITLE
MM-32782: RHSThread Follow Button

### PR DESCRIPTION
**Pull request by calebroseland**
#### Summary
Adds follow button in `RHSThread`. 

`actions/websocket_actions.jsx` changes overlap exactly with #7996 as it is required by both.

#### Ticket Link
https://mattermost.atlassian.net/browse/MM-32782

#### Screenshots
CleanShot 2021-04-26 at 17 16 24@2x

#### Release Note

```release-note
In Collapsed Reply Threads, added ability to follow thread from RHS
```

### CODE REVIEW

**Review by Willyfrog — APPROVED** [review]
LGTM

File: components/rhs_header_post/index.ts [hunk 1]
```diff
@@ -22,26 +28,29 @@ import {getIsRhsExpanded} from 'selectors/rhs';
...
+        currentTeamId: getCurrentTeamId(state),
+        currentUserId: getCurrentUserId(state),
+        isCollapsedThreadsEnabled: isCollapsedThreadsEnabled(state),
+        isFollowingThread: getThread(state, rootPostId)?.is_following,
     };
 }
 
-function mapDispatchToProps(dispatch: Dispatch) {
-    return {
-        actions: bindActionCreators({
```
Comment (jespino) [inline components/rhs_header_post/index.ts:34]: Why are we no longer binding the action creators?

Comment (calebroseland) [reply components/rhs_header_post/index.ts:34]: They're still bound, but with the flattened actions and passing them in, `connect` automatically does the binding.

**Review by jespino — APPROVED** [review]
Lgtm

**Comment by ogi-m** [comment]
@matthewbirtch @calebroseland Should the "Follow" button be hidden in the RHS for a root post with no replies?
Currently it's shown and clicking it doesn't do anything:

https://i.gyazo.com/09bec6c746274118e7e699ab98b43ee2.mp4

**Comment by matthewbirtch** [comment]
> @matthewbirtch @calebroseland Should the "Follow" button be hidden in the RHS for a root post with no replies?
> Currently it's shown and clicking it doesn't do anything:
> 
> https://i.gyazo.com/09bec6c746274118e7e699ab98b43ee2.mp4

I'm not sure about the technical implications, but from a user's standpoint, I would want to be able to follow a root post and have it added to my Global Threads view once replies come in. I believe that's how it was intended

**Comment by jgilliam17** [comment]
To add to Ogi's point when clicking on `Follow` on root post without replies 
a) `Follow` does not change to `Following` 
b) Selection is not remembered and added to Threads when replies come in

**Review by jgilliam17 — APPROVED** [review]
Thanks @calebroseland 
Tested, looks good to merge.
- Verified `Follow` button has been added to RHS reply thread. Additional functionality will be addressed in [PR 8107 ](https://github.com/mattermost/mattermost-webapp/pull/8107)

**Pull request merged**